clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -11207,6 +11207,10 @@ def warn_attribute_spelling_deprecated : Warning<
   InGroup<DeprecatedAttributes>;
 def note_spelling_suggestion : Note<
   "did you mean to use %0 instead?">;
+def warn_attribute_requires_non_negative_integer_argument : Warning<
+  "the resulting value of the %0 attribute %select{first|second|third}1"
```

Comment:
Possible alternative: we could also reuse the existing diagnostic text about implicit conversion, but put it under the new attribute grouping:
```
def warn_attribute_requires_non_negative_integer_argument : Warning<warn_impcast_integer_sign.Text>, InGroup<AcceptedAttributes>;
```
One oddity from this would be that users who pass `-Wno-sign-compare` will still get what looks like a sign comparison warning, but that's the same behavior as with the current patch anyway.

Reply:
Thanks @AaronBallman and @elizabethandrews for the comments on improving the diagnostic message. 

Does the new diagnostic should print  like this?
implicit conversion changes signedness: 'unsigned int' to ' int'

Example: [[intel::max_work_group_size(8, 8, -8)]]

Reply:
The types look backwards in the diagnostic. It's changing from `int` to `unsigned int`.

Also, you should check that the diagnostic points to the attribute argument that changes sign (the `-8`). Does the caret appear in a place where it's obvious what the diagnostic is talking about?

Btw, a second test would be `[[intel::max_work_group_size(-8, 8, -8)]]` to ensure that the diagnostic triggers twice (once for each `-8`) and is sufficiently clear.

Reply:
>>Also you should check that the diagnostic points to the attribute argument that changes sign (the -8).
Yes, it points to the place )-8)
>>Btw, a second test would be [[intel::max_work_group_size(-8, 8, -8)]] to ensure that the diagnostic triggers twice (once for each -8) and is sufficiently clear.
Yes, it triggers twice and correct place (once for each -8)

I have created https://github.com/intel/llvm/pull/2988 for this change.